LanguageFeatures/Inline-classes/static_analysis_inline_class_A02_t01.dart

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Assume that T1, .. Ts are types, and V resolves to an inline
+/// class declaration of the following form:
+///
+/// inline class V<X1 extends B1, .. Xs extends Bs> ... {
+///   final T id;
+///   V(this.id);
+///
+///   ... // Other members.
+/// }
+/// It is then allowed to use V<T1, .. Ts> as a type.
+///
+/// @description Checks that `V<T1, .. Ts>` can be used as a type
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=inline-class
+
+import "../../Utils/expect.dart";
+
+inline class V<X1 extends num, X2 extends Object> {
+  final X1 id;
+  V(this.id);
+
+  X2 foo() => id as X2;
+}
+
+main() {
+  V v1 = V(42);
+  Expect.isTrue(v1 is V<num, Object>);
+
+  V<int, num> v2 = V(42);
+  Expect.isTrue(v2 is V<int, num>);
```

Comment:
So what does it mean to test that it "can be used as a type"?

Checking that it is accepted at compile-time to have it in `is` and `as` is certainly relevant. (Do we also have a test which is about the dynamic semantics? E.g., we should have `v2 is V<num, Object>` as well, but also `v2 is V<num, Never>`, because the actual behavior is to test that `v2 is num`).

Another part of "can be used as a type" could be that it has the expect subtyping relations (unless that's already covered or planned for a different test). So `v2` should be assignable to a variable of type `V<num, Object>` (we expect covariance for all type parameters, as long as we don't have support for anything else), but not `V<int, Never>`.

Reply:
However, we should have some tests showing that the inline type can occur in other positions: Parameter and return types of function types, type variable bounds in classes and functions, and perhaps I forgot some more places.

I believe there would be a separate test for inline types as superinterfaces of anything-other-than-an-inline-class, expecting errors.

Reply:
> Do we also have a test which is about the dynamic semantics?

Not yet. [Dynamic semantics](https://github.com/dart-lang/language/blob/main/accepted/future-releases/inline-classes/feature-specification.md#dynamic-semantics-of-inline-classes) is a separate chapter of the spec. Will add such tests as tests for this chapter

> Another part of "can be used as a type" could be that it has the expect subtyping relations (unless that's already covered or planned for a different test).

I'll add these tests later as a Subtyping tests in a separate PR

> v2 is V<num, Never>

I'll add this

> However, we should have some tests showing that the inline type can occur in other positions

I'll add this